swift/ql/lib/codeql/swift/security/CleartextLoggingExtensions.qll

```diff
@@ -29,6 +29,18 @@ private class DefaultCleartextLoggingSink extends CleartextLoggingSink {
   DefaultCleartextLoggingSink() { sinkNode(this, "log-injection") }
 }
 
+/**
+ * An barrier for cleartext logging vulnerabilities.
+ *  - encryption; encrypted values are not cleartext.
+ *  - booleans; these are more likely to be settings, rather than actual sensitive data.
+ */
+private class CleartextLoggingDefaultBarrier extends CleartextLoggingBarrier {
+  CleartextLoggingDefaultBarrier() {
+    this.asExpr() instanceof EncryptedExpr or
+    this.asExpr().getType().getUnderlyingType() instanceof BoolType
+  }
+}
```

Comment:
@mathiasvp do you happen to know whether a potentially large set of barriers (here: all expressions of `Bool` type) is likely to be a performance problem or will only be evaluated in the restricted context of nodes on a potential flow path?  (if you're not sure, I will spend a bit of time investigating this)

Reply:
We've done similar things in C++ in the past (i.e., block flow through all integral-typed expressions), and performance for this has always been fine. So I'm not concerned about this at all.

Reply:
Awesome, thanks!